### PR TITLE
Feature/create-controller-and-routes-fields

### PR DESCRIPTION
**Pull request by JesusGo115**
## Summary by CodeRabbit

- **New Features**
    - Added functionality for CRUD operations on fields within groups, ensuring field name uniqueness and appropriate error handling.
    - Implemented routes for field operations using Express and a corresponding controller.

- **Bug Fixes**
    - Improved league name uniqueness checks to avoid duplicate entries during creation and updates.

- **Tests**
    - Introduced detailed tests for field-related routes, covering creation, updates, deletions, and error scenarios.

- **Documentation**
    - Defined validation rules for fields in the model to maintain data integrity and constraint adherence.

### CODE REVIEW

File: server/controllers/field.controller.js [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+        where: {
+          id: req.params['id'],
+        },
+      });
+
+      if (!currentField) {
+        throw new Error("Field with given ID was not found");
+      }
+
+      currentField.name = req.body.name || currentField.name;
```
Comment (saulRM10) [inline server/controllers/field.controller.js:76]: use Object.keys method

File: server/controllers/field.controller.js [hunk 2]
```diff
@@ -34,6 +34,8 @@ const FieldController = function () {
       },
     });
 
+    console.log(fieldFound);
```
Comment (saulRM10) [inline server/controllers/field.controller.js:37]: remove console log statement

File: server/controllers/field.controller.js [hunk 3]
```diff
@@ -70,20 +73,26 @@ const FieldController = function () {
...
-      currentField.width = req.body.width || currentField.width;
+      Object.keys(req.body).forEach(key => {
+        if (key !== "group_id"){
+          currentField[key] = req.body[key] ? req.body[key] : currentField[key];
+        }
+      });
 
       const fieldFound = await isNameUniqueWithinGroup(currentField.group_id, currentField.name, currentField.address, currentField.length, currentField.width);
 
+      console.log(fieldFound);
```
Comment (saulRM10) [inline server/controllers/field.controller.js:88]: remove console log statement

**Review by saulRM10 — APPROVED** [review]
second time reviewing it, remove some console logs statements you forgot to remove. Thanks for updating the controller with the feedback i left from the first review

File: server/controllers/field.controller.js (hunk 1 above)
Comment (JesusGo115) [reply server/controllers/field.controller.js]: Changed it to the Objects.keys method

**Pull request merged**